### PR TITLE
Show number of students while problem is open

### LINKED ISSUE #36: Show number of responses live as graph updates
When a problem is "Open", we will display the number of students who have responded in the upper right corner. After the problem is closed, this text disappears.

image

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #36

#### What's this PR do?
 - Adds a column to store the number of enrolled students at the time a problem was opened
 - Pass this information to the frontend. It will use the number of submissions and this to show a message to a staff user regarding the number of responses currently. The options in the run switcher will also now show this information for each run.

Note that existing runs will show 0 enrolled students, only new runs will have an accurate enrolled student count.

#### How should this be manually tested?
Open a new problem and submit a response. It should show 1 response in the text and some number of enrolled users. Close the problem and note that the select element has the new run with 1 response in the text.

### CODE REVIEW

File: rapid_response_xblock/static/js/src/rapid.js [hunk 1]
```diff
@@ -47,6 +48,28 @@
...
+      }
+
+      var totalCount = numResponses(runId);
+
+      var run = state.runs.find(function(run) {
+        return run.id === runId;
+      });
+      var numStudents = run ? run.num_enrolled : 0;
+
+      return totalCount + ' / ' + numStudents + ' students have answered';
```
[the remark this answers is not in the document]
Comment (noisecapella) [reply rapid_response_xblock/static/js/src/rapid.js]: @roberthouse54 Does that change work for you?

File: rapid_response_xblock/static/js/src/rapid.js [hunk 2]
```diff
@@ -328,10 +378,14 @@
...
         .text(function(run) {
+          var totalCount = numResponses(run.id);
+
           if (run.id === NONE_SELECTION) {
             return (chartIndex > 0) ? 'Select' : 'None';
           }
-          return moment(run.created).format("MMMM D, YYYY, h:mm:ss a");
+          var dateString = moment(run.created).format("MMMM D, YYYY, h:mm:ss a");
+
+          return dateString + " ------- " + totalCount + "/" + run.num_enrolled + " Students Responded";
```
[the remark this answers is not in the document]
Comment (noisecapella) [reply rapid_response_xblock/static/js/src/rapid.js]: @roberthouse54 What do you think?

**Comment by noisecapella** [comment]
I think that should be everything

File: rapid_response_xblock/static/js/src/rapid.js [hunk 3]
```diff
@@ -291,25 +307,49 @@
...
+     * @param {number} runId The run id
+     *
+     * @returns {number} The total number of responses
+     */
+    function numResponses(runId) {
+      var total = 0;
+      state.choices.forEach(function(item) {
+        total += state.counts[item.answer_id][runId] || 0;
+      });
+      return total;
```
Comment (gsidebo) [inline rapid_response_xblock/static/js/src/rapid.js]: j/w: Is this logic is going to change when the API returns the total counts (#57)?

File: rapid_response_xblock/static/js/src/rapid.js [hunk 4]
```diff
@@ -503,8 +552,16 @@
         pollMinutes = Math.floor(totalSeconds / 60);
       }
       $timer.text(pollMinutes + "m : " + pollSeconds + "s");
+      $timer.toggleClass("on", state.is_open);
       $timerSpinner.toggleClass('hidden', !state.is_open);
       $timerSpinnerText.toggleClass('hidden', !state.is_open);
+      $numStudents.toggleClass('hidden', !state.is_open);
+
+      // using chartIndex 0 since this message should only get shown if there is only one chart visible
+      if (state.runs.length > 0) {
```
Comment (gsidebo) [inline rapid_response_xblock/static/js/src/rapid.js]: Nit: You can probably get rid of the ternary operator below and just add it to this condition `if (state.runs.length > 0 && state.is_open)`. The element is already being hidden if the problem isn't open. You could move the `toggleClass` below this conditional so we don't show the element before the correct text is set. Won't hold up the PR for this, though.

File: rapid_response_xblock/static/js/src/rapid.js [hunk 5]
```diff
@@ -47,6 +48,21 @@
...
+     *
+     * @param {number} runId The run to create the message for
+     * @returns {string} The message
+     */
+    function makeNumStudentsMessage(runId) {
+      var totalCount = numResponses(runId);
+
+      var nounVerb = totalCount === 1 ? 'student has' : 'students have';
+
+      return totalCount + ' ' + nounVerb + ' answered';
```
Comment (gsidebo) [inline rapid_response_xblock/static/js/src/rapid.js]: Nit: still not crazy about the line breaks, but won't hold up the PR

File: rapid_response_xblock/static/js/src/rapid.js (hunk 3 above)
Comment (noisecapella) [reply rapid_response_xblock/static/js/src/rapid.js]: Yes it should

**Pull request merged**